src/main/java/com/google/sps/servlets/SearchServlet.java

```diff
@@ -223,8 +203,46 @@ private PreparedQuery queryProjectDatastore(HttpServletRequest request, Datastor
 
             }
 
+            Iterable<Entity> finalResults;
+
+            //Needs to return an iterable:
+            finalResults = results.asIterable();
+
+            if(!"".equals(searchBarText)){
+
+                //If there is text in the searchBar, perform a fuzzySearch in the queried projects:
+                finalResults = fuzzySearch(searchBarText, finalResults);
+
+            }
+
             //Return the results:
-            return results;
+            return finalResults;
+
+    }
+
+    //Filters the already filtered entities from the datastore query by it's title similarity to the searchBarText:
+    private Iterable<Entity> fuzzySearch(String searchText, Iterable<Entity> entities){
+
+        //0 is completely equal. 100 is equal:
+        int threshold = 30;
```

Comment:
usually it's best practice to place all constants as final static variables of the class. i.e.

private static final fuzzySearchThreshold = 30;

at the top of the class.

This way it's easy to find and modify these values later on if needed.

Reply:
Noted! It will be changed on my next commit.

Reply:
Also, my comment is wrong there. It should be 

> 0 is completely different. 100 is equal: